examples/module1/lesson1/solver/hooks/useCalculations.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+
+export function useCalculations() {
+  const [value1, setValue1] = useState<number>(0);
+  const [value2, setValue2] = useState<number>(0);
+  const [result, setResult] = useState<number | string>(0);
+  const [error, setError] = useState<string>('');
+
+  const calculate = (
+    func: (a: number, b: number) => number | [boolean, string | number]
```

Comment:
Ten wynik zwracany jest IMO niepotrzebnie złożony z dwóch typów. Spróbuj przyjąć, że każda funkcja będzie zwracać parę `[result, error]` i zobacz, jak zmieni się kod. Z jednej strony będzie to dodatkowa złożoność w funkcjach, gdzie na teraz nie przewidujesz błędów, ale z drugiej unikniesz tego żonglowania typami przez wyjątek, jakim jest dla ciebie dzielenie.

Reply:
Faktycznie sporo to uprościło, kosztem delikatnie większej złożoności w funkcjach.